carbonmark-data/app/[locale]/retirement-trends/page.tsx

```diff
@@ -0,0 +1,186 @@
+"use client";
+
+import { t } from "@lingui/macro";
+import TabContext from "@mui/lab/TabContext";
+import TabList from "@mui/lab/TabList";
+import TabPanel from "@mui/lab/TabPanel";
+import { Tab } from "@mui/material";
+import { MobileTabSelector } from "components/MobileTabSelector";
+import ChartCard from "components/cards/ChartCard";
+import DetailPage from "components/pages/DetailPage";
+import { FC, ReactNode, useState } from "react";
+import layout from "theme/layout.module.scss";
+import styles from "./styles.module.scss";
+
+const tabs = ["byPool", "byToken", "byChain", "byBeneficiary"] as const;
+type Tab = (typeof tabs)[number];
+
+function assertUnreachable(tab: never): never {
+  throw new Error("Unreachable code");
+}
+
+const TypedTabPanel: FC<{
+  tab: Tab;
+  children: ReactNode;
+  className: string;
+}> = ({ tab, children, className }) => {
+  switch (tab) {
+    case "byPool":
+    case "byToken":
+    case "byChain":
+    case "byBeneficiary":
+      return (
+        <TabPanel value={tab} className={className}>
+          {children}
+        </TabPanel>
+      );
+  }
+  return assertUnreachable(tab);
+};
+
+export default function RetirementTrends() {
+  const [activeTab, setActiveTab] = useState("byPool");
+
```

Comment:
Can you implement a way to select a tab through the URL ? We will need it for the back button of the associated detail pages.

Reply:
Done in [7c99612](https://github.com/KlimaDAO/klimadao/pull/1518/commits/7c99612cc211d886cc9667f1a330bdedc3b6d97c)

Reply:
I'd appreciate your feedback as I'm not sure how robust this solution is…

Reply:
Looks good to me, but I am not a React guru. I used to do it almost the same way with VueJS except that I used anchors instead of query parameters.
Those guys seem to do it like you: https://www.pluralsight.com/guides/handling-tabs-using-page-urls-and-react-router-doms

Reply:
nice, it's been working flawlessly in my tests